src/sentry/tasks/deletion.py

```diff
@@ -129,7 +139,13 @@ def delete_project(object_id, transaction_id=None, continuous=True, **kwargs):
         p.update(status=ProjectStatus.DELETION_IN_PROGRESS)
 
     # Immediately revoke keys
-    ProjectKey.objects.filter(project_id=object_id).delete()
+    project_keys = list(ProjectKey.objects.filter(project_id=object_id).values_list('id', flat=True))
+    ProjectKey.objects.filter(id__in=project_keys).delete()
```

Comment:
Alternatively, you could just also do `ProjectKey.objects.filter(project_id=object_id).delete()` and deal with the fact that we maybe miss logging a row. But I don't have a strong preference.